### PR TITLE
Adds support for registering client dial option handlers

### PR DESCRIPTION
**Pull request by mohit-a21**

### CODE REVIEW

File: config/configgrpc/configgrpc.go [hunk 1]
```diff
@@ -311,3 +333,11 @@ func GetGRPCCompressionKey(compressionType string) string {
 	}
 	return CompressionUnsupported
 }
+
+func RegisterClientDialOptionHandlers(handlers ...ClientDialOptionHandler) {
+	mu.Lock()
```
Comment (pavolloffay) [inline config/configgrpc/configgrpc.go]: why do we need mutex here?

File: config/configgrpc/configgrpc.go [hunk 2]
```diff
@@ -40,13 +41,20 @@ const (
 	PerRPCAuthTypeBearer = "bearer"
 )
 
+type ClientDialOptionHandler func() grpc.DialOption
```
Comment (tim-mwangi) [inline config/configgrpc/configgrpc.go:43]: Should we contribute this back? CC @pavolloffay

File: config/configgrpc/configgrpc.go (hunk 1 above)
Comment (mohit-a21) [reply config/configgrpc/configgrpc.go]: Can we guarantee that this func is called only from a single thread?

Comment (mohit-a21) [reply config/configgrpc/configgrpc.go]: Removed the mutexes.

**Review by davexroth — APPROVED** [review]

**Pull request merged**